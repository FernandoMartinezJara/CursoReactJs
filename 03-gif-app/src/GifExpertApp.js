import {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import {GifGrid} from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = []}) => {

    const [categories, setcategories] = useState(defaultCategories);

    return <>
        <h2>GifExpertApp</h2>

        <AddCategory 
            setcategories={setcategories}/>

        <hr/>

        <ul>
            {
                categories.map((cat, i) => {

                   return <GifGrid 
                        key={cat}
                        category = {cat} />
                })
            }
        </ul>

        

    </>
}

export default GifExpertApp;