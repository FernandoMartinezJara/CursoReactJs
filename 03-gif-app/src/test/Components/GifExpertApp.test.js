import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import GifExpertApp from '../../GifExpertApp';

describe('Pruebas componente <GifExpertApp/>', () => {

    test('debe de mostrar correctamente', () => {

        const wrapper = shallow(<GifExpertApp/>);

        expect(wrapper).toMatchSnapshot();
        
    });
    
    test('debe de mostrar una lista de categorias', () => {

        const categories = ['Bon Jovi', 'Guns n Roses'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('GifGrid').length).toBe(categories.length);


        
    })
    
    
})
