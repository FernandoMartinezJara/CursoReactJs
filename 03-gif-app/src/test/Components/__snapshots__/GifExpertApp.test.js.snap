// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente <GifExpertApp/> debe de mostrar correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Pruebas componente <GifExpertApp/> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setcategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="Bon Jovi"
      key="Bon Jovi"
    />
    <GifGrid
      category="Guns n Roses"
      key="Guns n Roses"
    />
  </ul>
</Fragment>
`;
